litgpt/pretrain.py

```diff
@@ -83,20 +82,39 @@ def setup(
         train: Training-related arguments. See ``litgpt.args.TrainArgs`` for details.
         eval: Evaluation-related arguments. See ``litgpt.args.EvalArgs`` for details.
         optimizer: An optimizer name (such as "AdamW") or config.
-        
+
         devices: How many devices/GPUs to use. Uses all GPUs by default.
         tokenizer_dir: Optional path to the tokenizer dir that was used for preprocessing the dataset. Only some data
             module require this.
         logger_name: The name of the logger to send metrics to.
         seed: The random seed to use for reproducibility.
     """
+    if model_name == "list":
+        available_models = "\n".join(sorted(name_to_config))
+        print(f"Available values:\n{available_models}")
+        quit()
+
+    # Support both model_name options: meta-llama/Meta-Llama-3-8B & Meta-Llama-3-8B
+    if model_config is None:
+        num_slashes = model_name.count("/")
+        if num_slashes > 1:
+            raise ValueError(
+                f"Invalid model_name: {model_name}. The model_name argument cannot contain multiple slashes."
+                " It should be either in the format 'meta-llama/Meta-Llama-3-8B' or 'Meta-Llama-3-8B'."
+                )
+        if num_slashes:
+            try:
+                model_name = Config.get_name_from_hf_config(model_name)
+            except ValueError:
+                available_models = "\n".join(sorted(name_to_config))
+                print(f"Available values:\n{available_models}")
+                quit()
+    else:
+        model_name = "placeholder"
+
     hparams = capture_hparams()
     data = TinyLlama() if data is None else data
-    if model_config is not None and model_name is not None:
-        raise ValueError("Only one of `model_name` or `model_config` can be set.")
```

Comment:
How do we provide a config now? I think now you'd have to set both model_name and config. Can you update the pretraining config files?

Reply:
Good point, should be addressed.